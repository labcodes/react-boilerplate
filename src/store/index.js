import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';

import actions from '../actions';

// ====

const store = createStore(
    actions,
    applyMiddleware(logger)
);

// ====

export default store;
