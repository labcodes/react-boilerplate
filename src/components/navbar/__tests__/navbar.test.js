'use strict';

import React from 'react';

// Test helpers
import renderer from 'react-test-renderer';

// Components connected
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from 'store';

// Component
import Navbar from 'components/navbar';

// ====

describe('Navbar [Snapshot]', () => {
    it('Renders correctly', () => {
        const component = (
            <BrowserRouter>
                <Provider store={store}>
                    <Navbar />
                </Provider>
            </BrowserRouter>
        );

        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
