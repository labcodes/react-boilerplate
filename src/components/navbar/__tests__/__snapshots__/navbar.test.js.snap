// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar [Snapshot] Renders correctly 1`] = `
<aside
  className="sc-bdVaJa dwtwig"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
</aside>
`;
